core/code/entity_decode.js

```diff
@@ -60,10 +60,20 @@ function parseArtifactDetail(arr) {
   };
 }
 
+function parseHistoryDetail(bitarray) {
+  if (bitarray == null) { return null; }
+  return {
+    _raw: bitarray,
+    visited:  !!(bitarray & (1|2|4)),
```

Comment:
Visited should just be:
```suggestion
    visited:  !!(bitarray & 1),
```
There is no guarantee that scanned = visited for historical data, and the bit will always be set when appropriate.

Reply:
> There is no guarantee that scanned = visited for historical data

Could you explain what may be different (and why) for historical data?

Reply:
P.S.
In either case, even if we decide to fix (supposed) intel bug and make correction on our side, let's do that explicitly.
Like
```js
var parsed = {
  visited:  !!(bitarray & 1),`
  //
};
if (parsed.captured || parsed.scoutControlled) {
  // fix (supposed) intel bug on our side
  parsed.visited = true;
}

Reply:
Attention!
At the moment 'scanned' layer is wrongly titled as 'Scout Controlled' in stock intel!
"Scout Controller" is a volatile status, that the agent looses sooner or later on interaction of other agents with the same portal.
"scanned" is a permanent status, set when the agent submits the first 3D-Scan of a portal.

Reply:
I think we better leave that as an operation on the bitarray for speed reasons, but add a comment.

Reply:
The point is you're making an unnecessary decision for the purposes of what?  Intentionally making mistakes?

If a scout controlled portal is always visited, the first bit will always be set.
If a captured portal is always visited, the first bit will always be set.

You will never need a "catchup method" by setting it when the others are true, and all you're doing is introducing a bug, then claiming it's irrelevant.